src/gjk/power_to_energy.py

```diff
@@ -0,0 +1,173 @@
+from gjk.models import ReadingSql
+from gjk.models import DataChannelSql
+from gjk.config import Settings
+from sqlalchemy.orm import Session, sessionmaker
+from sqlalchemy.orm import declarative_base
+from sqlalchemy import create_engine
+from sqlalchemy import asc 
+from sqlalchemy import Column, BigInteger, Integer, String, Float
+from sqlalchemy import UniqueConstraint
+import pendulum
+import dotenv
+from datetime import datetime
+import pandas as pd
+import os
+
+# ------------------------------------------
+# User inputs
+# ------------------------------------------
+
+timezone = 'America/New_York'
+start= pendulum.datetime(2024, 1, 1, 0, 0, tz=timezone)
+end = pendulum.datetime(2024, 2, 2, 20, 0, tz=timezone)
+
+# Record 0 Wh values in the energy table
+record_zeros = False
+
+# ------------------------------------------
+# Import readings data
+# ------------------------------------------
+
+dotenv.load_dotenv()
+engine = create_engine(os.getenv('GJK_DB_URL'))
+Session = sessionmaker(bind=engine)
+session = Session()
+
+saved_sql_channels = session.query(DataChannelSql).all()
+saved_channel_ids = [channel.id for channel in saved_sql_channels]
+
+# Convert to Unix timestamp in milliseconds
+start_ms = int(start.add(hours=-1).timestamp() * 1000)
+end_ms = int(end.timestamp() * 1000)
+
+readings = session.query(ReadingSql).filter(
+    ReadingSql.time_ms >= start_ms,
+    ReadingSql.time_ms < end_ms,
+).order_by(asc(ReadingSql.time_ms)).all()
+
+# ------------------------------------------
+# Split the data by channel
+# ------------------------------------------
+
+power_channels = ['hp-idu', 'hp-odu', 'dist-pump', 'primary-pump', 'store-pump']
```

Comment:
Avoid hard-coding strings like this. We might end up having other channels that include hp-idu for example. 

power_channels = list(filter(lambda x: x.telemetry_name == TelemetryName.PowerW.value, channels))
channel_names = [channel.name for channel in power_channels]

Reply:
At the end of your code line, what is `channels`?